src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -109,23 +132,118 @@ private static synchronized void closeResources()  {
       zoneIdToTable.clear();
       zoneIdToTable = null;
     }
-    if (fixedTransitions != null) {
-      fixedTransitions.close();
-      fixedTransitions = null;
+    if (transitions != null) {
+      transitions.close();
+      transitions = null;
     }
   }
 
-  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
-    // TODO: Remove this check when all timezones are supported
-    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
-    if (!isSupportedTimeZone(currentTimeZone)) {
-      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
-          currentTimeZone.toString()));
+  private static long getScaleFactor(ColumnVector input){
+    DType inputType = input.getType();
+    if (inputType == DType.TIMESTAMP_SECONDS){
+      return 1;
+    } else if (inputType == DType.TIMESTAMP_MILLISECONDS){
+      return 1000;
+    } else if (inputType == DType.TIMESTAMP_MICROSECONDS){
+      return 1000*1000;
+    }
+    throw new UnsupportedOperationException("Unsupported data type: " + inputType);
+  }
+
+  public static boolean isSupportedTimeZone(String zoneId) {
+    try {
+      getZoneId(zoneId); // check that zoneId is a valid zone
+      return true;
+    } catch (ZoneRulesException e) {
+      return false;
+    }
+  }
+
+  /**
+   * @deprecated added to not break spark-rapids build
+   * Will be removed after spark-rapids changes are merged,
+   * since all timezones will be supported.
+   * This function will not be called after spark-rapids changes
+   */
+  public static boolean isSupportedTimeZone(ZoneId desiredTimeZone) {
+    return desiredTimeZone != null &&
+      (desiredTimeZone.getRules().isFixedOffset() ||
+      desiredTimeZone.getRules().getTransitionRules().isEmpty());
+  }
+
+  // enforce that all timestamps, regardless of timezone, be less than the desired date
+  private static boolean shouldFallbackToCpu(ColumnVector input, ZoneId zoneId){
```

Comment:
For the non Daylight Saving timezone(DST) : tz.getRules().getTransitionRules().isEmpty(), if there are timestamp > fall_back_limitation, it will use CPU. But original behavior is to compute on GPU.
Although it's rare case, but it does changed the original behavior.
And the extra out-of-range checking will slow down the speed for non-DST tzs.

Reply:
When it's non-DST and there is value > 2200 year, now it runs on CPU.
But before it runs on GPU, do we need to chang this behavior?

Reply:
Yes, this is done.